### ISSUE TITLE
@EnableConfigServer not creating the ResourceRepository bean

### ISSUE THREAD

**Issue by duartesantos** [spring-cloud/spring-cloud-config#549]
My business service is requesting binaries through the config server using the `ResourceController`, but since I updated to version 1.2.1.RELEASE it stopped working.
I dug to understand what was happening and I realised that the `ResourceRepositoryConfiguration` is not creating the `ResourceRepository` bean. The `ResourceRepositoryConfiguration` has a creation condition if `SearchPathLocator` exists, which is only by `EnvironmentRepositoryConfiguration`.

Basically I had to change the order which the configurations are bing imported by the `@EnableConfigServer` to:
`@Import({ EnvironmentRepositoryConfiguration.class, ResourceRepositoryConfiguration.class,
		ConfigServerEncryptionConfiguration.class, ConfigServerMvcConfiguration.class })`

**Comment by ryanjbaxter** [comment]
Probably due to the changes from @spencergibb in this commit
https://github.com/spring-cloud/spring-cloud-config/commit/465bc949534d1ad48bfa20ae5ba947803796a5e9

The order in your issue appears to be the same as what was [released](https://github.com/spring-cloud/spring-cloud-config/blob/465bc949534d1ad48bfa20ae5ba947803796a5e9/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/ConfigServerAutoConfiguration.java#L31https://github.com/spring-cloud/spring-cloud-config/blob/465bc949534d1ad48bfa20ae5ba947803796a5e9/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/ConfigServerAutoConfiguration.java#L31).

The original order was
```
@Import({ ResourceRepositoryConfiguration.class, EnvironmentRepositoryConfiguration.class, 
ConfigServerEncryptionConfiguration.class, ConfigServerMvcConfiguration.class })
```

**Comment by duartesantos** [comment]
@ryanjbaxter You're right. Sorry for the repetition. I can see this issue was already referred in:
https://github.com/spring-cloud/spring-cloud-config/issues/530
https://github.com/spring-cloud/spring-cloud-config/issues/524
Closing this issue.
